apps/frontend/assets/src/omg-2fa-modal/Enable2FaModal.js

```diff
@@ -0,0 +1,157 @@
+
+const Create2FaModalContainer = styled.div`
+  padding: 40px;
+  text-align: center;
+  width: 400px;
+  position: relative;
+  i {
+    position: absolute;
+    top: 15px;
+    right: 15px;
+    cursor: pointer;
+  }
+  button {
+    margin-top: 20px;
+  }
+  h4 {
+    margin-bottom: 10px;
+  }
+  input {
+    text-align: center;
+    font-size: 18px;
+  }
+  .backup-container {
+    text-align: center;
+    p {
+      margin: 20px 0;
+    }
+  }
+  .backup-item {
+    padding: 5px;
+    display: inline-block;
+    width: 110px;
+  }
+`
+
+function CreateTwoFaModal ({ open, onRequestClose, history }) {
+  const dispatch = useDispatch()
+  const [secretCode, setSecretCode] = useState(null)
+  const [backupCodes, setBackupCodes] = useState(null)
+  const [passcode, setPasscode] = useState('')
+  const [submitStatus, setSubmitStatus] = useState('DEFAULT')
+
+  const onEnable2Fa = async () => {
+    const { data: backupCodeResult } = await createBackupCodes()(dispatch)
+    if (backupCodeResult) {
+      const { data: secretResult } = await enable2Fa(passcode)(dispatch)
+      if (secretResult) {
+        setBackupCodes(backupCodeResult)
+        setSubmitStatus('SUCCESS')
+      }
+    } else {
+      setSubmitStatus('FAILED')
+    }
+  }
+  const afterClose = () => {
+    setPasscode('')
+    setSecretCode(null)
+    setBackupCodes('')
+    setSubmitStatus('DEFAULT')
+  }
+  const onAfterOpen = () => {
+    createSecretCodes()(dispatch).then(({ data }) => {
+      if (data) setSecretCode(data)
+    })
+  }
+
+  const onSubmit = e => {
+    e.preventDefault()
+    setSubmitStatus('LOADING')
+    onEnable2Fa()
+  }
+
+  const onOkBackupCode = () => {
+    onRequestClose()
+    history.push('/login')
+  }
+
+  const renderCreateMode = () => {
+    return (
+      <form onSubmit={onSubmit}>
+        <h4>please scan the QR</h4>
+        <div>
+          Secret code: {_.get(secretCode, 'secret_2fa_code', 'loading code..')}
+        </div>
+        <QrCode data={secretCode && to2FaFormat(secretCode)} size={200} />
+        <Input
+          value={passcode}
+          onChange={e => setPasscode(e.target.value)}
+          normalPlaceholder='passcode...'
+        />
+        <Button loading={submitStatus === 'LOADING'}>
+          Enable Two Factor Authentication
+        </Button>
+      </form>
+    )
+  }
+
+  const renderShowBackup = () => {
+    return (
+      <div className='backup-container'>
+        <h4>Please keep your backup code</h4>
+        <p>
+          You can print or download one-time use backup codes for times when
+          your phones are unavailable.
+        </p>
+        {backupCodes.backup_codes.map(backupCode => {
+          return (
+            <div key={backupCode} className='backup-item'>
+              {backupCode}
+            </div>
+          )
+        })}
+        <div>
+          <Button onClick={onOkBackupCode}>
+            {'Ok, I\'ve saved the backup codes'}
+            <br /> ( This will log you out )
```

Comment:
you can still close the modal at this step, and then you would navigate an empty app with 'invalid auth token' errors, so you should logout onrequestclose as well